File: tests/cpp/MultifacetedAbstractionTest/NotMultifacetedAbstractionTest3.cpp
// The class has more than 6 fields and more than 6 methods, but LCOM lower than 0.8.
// -> Has not multifaceted abstraction smell.
#include "NotMultifacetedAbstractionTest3.h"

void NotMultifacetedAbstractionTest3 :: func1() {
    int v = this->f1;
}
void NotMultifacetedAbstractionTest3 :: func2() {
    this->func1();
}
void NotMultifacetedAbstractionTest3 :: func3() {
    int v = this->f1;
}
void NotMultifacetedAbstractionTest3 :: func4() {
    this->func1();
}

File: tests/java/MultifacetedAbstractionTest/NotMultifacetedAbstractionTest3.java
// The class has more than 6 fields and more than 6 methods, but LCOM lower than 0.8.
// -> Has not multifaceted abstraction smell.
public class NotMultifacetedAbstractionTest3 //Type originates from: NotMultifacetedAbstractionTest3.h
{
	private int f1;
	private int f2;
	private int f3;
	private int f4;
	private int f5;
	private int f6;
	private int f7;
	private void func1() //Method definition originates from: NotMultifacetedAbstractionTest3.cpp
	{
		int v = this.f1;
	}

	private void func2() //Method definition originates from: NotMultifacetedAbstractionTest3.cpp
	{
		this.func1();
	}

	private void func3() //Method definition originates from: NotMultifacetedAbstractionTest3.cpp
	{
		int v = this.f1;
	}

	private void func4() //Method definition originates from: NotMultifacetedAbstractionTest3.cpp
	{
		this.func1();
	}

	private void func5()
	{
	}
	private void func6()
	{
	}
	private void func7()
	{
	}
	private void func8()
	{
	}
	private void func9()
	{
	}
	private void func10()
	{
	}

}
